### PR TITLE
[DS-350] Added slot function to hopper

### PR DESCRIPTION
**Pull request by vicky-comeau**

### CODE REVIEW

File: packages/components/src/IconList/src/IconList.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ function IconList(props:IconListProps, ref: ForwardedRef<HTMLSpanElement>) {
  *
  * [View Documentation](TODO)
  */
-const _IconList = forwardRef<HTMLSpanElement, IconListProps>(IconList);
+const _IconList = slot("icon", forwardRef<HTMLSpanElement, IconListProps>(IconList));
```
Comment (alexasselin008) [inline packages/components/src/IconList/src/IconList.tsx:66]: slot as slotFn?

File: packages/components/src/buttons/src/ButtonGroup.tsx [hunk 2]
```diff
@@ -117,7 +117,7 @@ function ButtonGroup(props: ButtonGroupProps, ref: ForwardedRef<HTMLDivElement>)
  *
  * [View Documentation](TODO)
  */
-const _ButtonGroup = forwardRef(ButtonGroup);
+const _ButtonGroup = slot("button-group", forwardRef(ButtonGroup));
```
Comment (alexasselin008) [inline packages/components/src/buttons/src/ButtonGroup.tsx:120]: slot as slotFn

File: packages/components/src/tag/src/Tag.tsx [hunk 3]
```diff
@@ -192,7 +193,7 @@ function Tag(props: TagProps, ref: ForwardedRef<HTMLDivElement>) {
  *
  * [View Documentation](TODO)
  */
-const _Tag = forwardRef<HTMLDivElement, TagProps>(Tag);
+const _Tag = slot("tag", forwardRef<HTMLDivElement, TagProps>(Tag));
```
Comment (alexasselin008) [inline packages/components/src/tag/src/Tag.tsx:196]: slot as slotFn

File: packages/components/src/typography/Heading/src/Heading.tsx [hunk 4]
```diff
@@ -64,7 +64,7 @@ function Heading(props: HeadingProps, ref: ForwardedRef<HTMLHeadingElement>) {
  *
  * [View Documentation](TODO)
  */
-const _Heading = forwardRef<HTMLHeadingElement, HeadingProps>(Heading);
+const _Heading = slot("heading", forwardRef<HTMLHeadingElement, HeadingProps>(Heading));
```
Comment (alexasselin008) [inline packages/components/src/typography/Heading/src/Heading.tsx:67]: slot as slotFn

File: packages/components/src/typography/Text/src/Text.tsx [hunk 5]
```diff
@@ -75,7 +76,7 @@ function Text(props: TextProps, ref: ForwardedRef<HTMLSpanElement>) {
  *
  * [View Documentation](TODO)
  */
-const _Text = forwardRef<HTMLSpanElement, TextProps>(Text);
+const _Text = slot("text", forwardRef<HTMLSpanElement, TextProps>(Text));
```
Comment (alexasselin008) [inline packages/components/src/typography/Text/src/Text.tsx:79]: slot as slotFn

File: packages/icons/src/Icon.tsx [hunk 6]
```diff
@@ -86,4 +86,7 @@ export const Icon = forwardRef<SVGSVGElement, IconProps>((props, ref) => {
     );
 });
 
-Icon.displayName = "Icon";
+const _Icon = slot("avatar", Icon);
```
Comment (victortrinh2) [inline packages/icons/src/Icon.tsx:89]: [Question] Is it called "avatar" on purpose?

Comment (alexasselin008) [inline packages/icons/src/Icon.tsx:89]: slot as slotFn

Comment (alexasselin008) [inline packages/icons/src/Icon.tsx:89]: wrong slot?

Comment (vicky-comeau) [reply packages/icons/src/Icon.tsx:89]: fudge, good catch :P

Comment (vicky-comeau) [reply packages/icons/src/Icon.tsx:89]: no :(

**Review by alexasselin008 — DISMISSED** [review]

**Review by victortrinh2 — APPROVED** [review]

**Pull request merged**